From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Benedikt Niedermayr <benedikt.niedermayr@siemens.com>
Date: Fri, 12 Apr 2024 10:31:32 +0200
Subject: [PATCH] gpio-chardev: fix ressource handling

- "path" should only be freed in else statement.
- close "fh" only if not already allocated/opened

Signed-off-by: Benedikt Niedermayr <benedikt.niedermayr@siemens.com>
---
 src/gpio/gpio_chardev.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/gpio/gpio_chardev.c b/src/gpio/gpio_chardev.c
index 2cd159680010..e8ff4f3046d4 100644
--- a/src/gpio/gpio_chardev.c
+++ b/src/gpio/gpio_chardev.c
@@ -266,7 +266,7 @@ mraa_get_chip_base_by_number(unsigned number)
     path = mraa_file_unglob(glob);
     free(glob);
     if (!path) {
-        syslog(LOG_ERR, "[GPIOD_INTERFACE]: invalid chip number");
+        syslog(LOG_ERR, "[GPIOD_INTERFACE]: invalid chip number: %d", number);
         res = -1;
     } else {
         fh = fopen(path, "r");
@@ -274,9 +274,11 @@ mraa_get_chip_base_by_number(unsigned number)
             syslog(LOG_ERR, "[GPIOD_INTERFACE]: could not read from %s", path);
             res = -1;
         }
+        free(path);
     }
 
-    free(path);
+    if(fh)
+        fclose(fh);
     return res;
 #endif
 }
